src/components/shared/forms/ProjectDetailsForm.tsx

```diff
@@ -0,0 +1,77 @@
+import { Button, Form, FormInstance } from 'antd'
+import { t, Trans } from '@lingui/macro'
+
+import { FormItems } from 'components/shared/formItems'
+import { normalizeHandle } from 'utils/formatHandle'
+import { cidFromUrl, unpinIpfsFileByCid } from 'utils/ipfs'
+
+export type ProjectDetailsFormFields = {
+  name: string
+  description: string
+  infoUri: string
+  handle: string
+  logoUri: string
+  twitter: string
+  discord: string
+  payButton: string
+  payDisclosure: string
+}
+
+export default function ProjectDetailsForm({
+  form,
+  onFinish,
+  hideProjectHandle = false,
+}: {
+  form: FormInstance<ProjectDetailsFormFields>
+  onFinish: (values: ProjectDetailsFormFields) => void
+  hideProjectHandle?: boolean
+}) {
+  return (
+    <Form form={form} layout="vertical" onFinish={onFinish}>
+      <FormItems.ProjectName
+        name="name"
+        formItemProps={{
+          rules: [{ required: true }],
+        }}
+        onChange={name => {
+          const val = name ? normalizeHandle(name) : ''
+          // Use `handle` state to enable ProjectHandle to validate while typing
+          form.setFieldsValue({ handle: val })
+        }}
+      />
+      {!hideProjectHandle && (
```

Comment:
V2 doesn't use `handle`'s. This gives us the ability to hide the handle input for V2.